### PR TITLE
add bayescan tool

### PR DESCRIPTION
**Pull request by cmonjeau**

### CODE REVIEW

File: tools/bayescan/bayescan.xml [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  <description>Detecting natural selection from population-based genetic data</description>
+  <requirements>
+         <requirement type="package" version="2.0.1">bayescan</requirement>
+  </requirements>
+  <command>
+
+  <![CDATA[
+  mkdir "output_dir";
+
+  bayescan2 $input
```
Comment (bgruening) [inline tools/bayescan/bayescan.xml]: all files and text parameters should be single quoted.

File: tools/bayescan/bayescan.xml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+        <param name="input" type="data" format="tabular,txt" label="Input genotype data file" help="must be space/tab delimitted plain text file" />
+        <conditional name="loci_file">
+        <param name="loci" type="select" label="Discard loci?" help="" >
+                <option value="0">No</option>
+                <option value="1">Yes</option>
+        </param>
+        <when value="0"></when>
+        <when value="1">
+                        <param name="input_loci" type="data" format="tabular,txt" label="Discard loci file" help="Optional input file containing list of loci to discard" />
```
Comment (bgruening) [inline tools/bayescan/bayescan.xml]: Can you please improve the indentation a little bit? We recommend to use 4-space indentation.

File: tools/bayescan/bayescan.xml [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        <param name="num_pilot_runs" type="integer" value="20" label="Number of pilot runs" />
+        <param name="length_pilot_run" type="integer" value="5000" label="Length of pilot runs" />
+        <param name="burn" type="integer" value="50000" label="Additional burn-in length" help="Nlength of an interval between two draws in MCMC"/>
+
+        <param name="prior_odds" type="integer" value="10" size="3" label="Prior odds for the neutral model"/>
+        <param name="lower_prior" type="float" value="0.0" label="Lower bound for uniform prior on Fis (dominant data)"/>
+        <param name="higher_prior" type="float" value="1.0" label="Higher bound for uniform prior on Fis (dominant data)"/>
+        <param name="threshold" type="float" value="0.1" label="Threshold for the recessive genotype as a fraction of maximum band intensity"/>
+
+        <param name="pilot_runs" type="boolean" checked="false" label="Optional output file for pilot runs"/>
```
Comment (bgruening) [inline tools/bayescan/bayescan.xml]: If you use truevalue and falsevalue here you can save a little bit of code in the cheetah section.

File: tools/bayescan/bayescan.xml [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+Therefore, this formulation can consider realistic ecological scenarios where the effective size and the immigration rate may differ among subpopulations.
+
+.. _Bayescan: http://cmpg.unibe.ch/software/BayeScan/
+
+
+]]>
+
+</help>
+<citations>
+   <citation type="doi">10.1534/genetics.108.092221</citation>
```
Comment (bgruening) [inline tools/bayescan/bayescan.xml]: Are all 3 references really needed?

**Comment by cmonjeau** [comment]
@bgruening Thanks a lot for your advices. I'm a little rust on Galaxy tools creation.

**Comment by abretaud** [comment]
Green now! @bgruening should we try to remove the travis_wait before merging?

**Comment by bgruening** [comment]
@abretaud I don't think this travis wait hurts, we can leave this in.
@cmonjeau can we add into the help which program produces the input to this tool here. The datatypes are really generic and people might wonder how to generate this input file.

File: tools/bayescan/bayescan.xml [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+                <option value="1">Yes</option>
+            </param>
+            <when value="0"></when>
+            <when value="1">
+                <param name="input_loci" type="data" format="tabular,txt" label="Discard loci file" help="Optional input file containing list of loci to discard" />
+            </when>
+        </conditional>
+        <param name="snp_genotypes_matrix" type="boolean" checked="false" truevalue="-fstat" falsevalue="" label="SNP genotypes matrix data" help="Use SNP genotypes matrix"/>
+
+        <param name="output_prefix" type="text" value="bayescan" label="Output file prefix" />
```
Comment (nsoranzo) [inline tools/bayescan/bayescan.xml]: Does the tool fail if this is empty? You may want to add a validator:
```
<validator type="empty_field" />
```

Comment (nsoranzo) [inline tools/bayescan/bayescan.xml]: Or better, you can remove this altogether, user should not need to fill this.

Comment (cmonjeau) [inline tools/bayescan/bayescan.xml]: Indeed :). Thanks

**Comment by bgruening** [comment]
Thanks @cmonjeau, @abretaud and @nsoranzo!

**Pull request merged**